src/app/header/header.component.ts

```diff
@@ -35,35 +38,34 @@ export class HeaderComponent implements OnInit {
         this.navbarHidingFlag = false;
       }
     });
-    this.sources.forEach(Element => {
-      this.sourcesSet.add(Element.category);
-    });
   }
 
   onChangeUpdate(value: string) {
-    const selectedSource = this.sources.find(
-      source => source.category === value
-    );
-    this.headerLabel = selectedSource
-      ? selectedSource.category
-      : "Welcome to NewsFeeds";
+    if (value === "All Sources") {
+      this.updateDataService.updateHeading("Welcome to NewsFeeds");
+    } else {
+      this.updateDataService.updateHeading(value);
+    }
     this.updateDataService.updateSource(value);
   }
 
   filterArticles() {
     let filterPattern = this.filterLabel;
     this.filterLabel = "";
+    this.updateDataService.filterArticles(filterPattern);
     if (filterPattern != "") {
-      this.headerLabel = 'Search Result for "' + filterPattern + '"';
+      this.updateDataService.updateHeading(
+        'Search Result for "' + filterPattern + '"'
```

Comment:
You can ES6 string literals syntax instead of using the `+` operator for string concatenation.

``` `Search Result for ${filterPattern}` ```

Reply:
**Changes Implemented**